addons/point_of_sale/static/src/app/screens/ticket_screen/ticket_screen.js

```diff
@@ -144,31 +144,21 @@ export class TicketScreen extends Component {
                 this.state.selectedOrder = null;
             }
         }
-        order.uiState.displayed = false;
-        if (order && (await this._onBeforeDeleteOrder(order))) {
-            if (Object.keys(order.last_order_preparation_change).length > 0) {
-                await this.pos.sendOrderInPreparationUpdateLastChange(order, true);
-            }
-            if (order === this.pos.get_order()) {
-                this._selectNextOrder(order);
-            }
-            const cancelled = this.pos.removeOrder(order, true);
-
-            const idToCancel = [...this.pos.pendingOrder.delete];
 
-            if (idToCancel.length > 0) {
-                this.pos.pendingOrder.delete.clear();
-                await this.pos.data.call("pos.order", "action_pos_order_cancel", idToCancel);
-            }
-
-            if (!cancelled) {
-                order.uiState.displayed = true;
-            }
+        order.uiState.displayed = false;
+        if (order === this.pos.get_order()) {
```

Comment:
```suggestion
        if (order.id === this.pos.get_order().id) {
```

I remembered that referential equality doesn't work if the object is wrapped with the reactivity proxy.